tests/containers/bci_repo.pm

```diff
@@ -42,53 +42,81 @@ my $tdata = [
+    # refresh services in order to remove container-suseconnect-zypp orphaned services
+    container_exec($container, 'zypper refs');
+    # remove SLE_BCI repo pointing to official update servers and add SUT repo
+    container_exec($container, 'zypper rr SLE_BCI');
+    container_exec($container, "zypper ar http://openqa.suse.de/assets/repo/$bci_repo BCI_TEST");
+    container_exec($container, 'zypper ref', timeout => 180);
+    container_exec($container, 'zypper lr -d');
+}
+
+my $errors = 0;
+
 sub run {
     my ($self) = @_;
     $self->select_serial_terminal;
-    my $bci_repo = get_required_var('REPO_BCI');
 
     # download and start a BCI container
     my $image = get_required_var('CONTAINER_IMAGE_TO_TEST');
     record_info('IMAGE', $image);
     script_retry("podman pull $image", timeout => 300, delay => 60, retry => 3);
     record_info('Inspect', script_output("podman inspect $image"));
 
+    record_info('TEST', 'Test that the repo does not contain certain packages such as kernel, yast, desktop, kvm, etc...');
+    my $container = "bci-repo-tester_pkgs";
+    assert_script_run("podman run --name $container -dt $image");
+    prepare_repo($container);
+    foreach my $pkg (@pkg_regex) {
+        my $result = script_run(qq[podman exec $container /bin/sh -c 'zypper se $pkg'], die_on_timeout => 1);
+        if ($result == 0) {
+            # Fail if pkg is present in the repo.
+            record_soft_failure("poo#109822 - Package $pkg should not be present in BCI-repo!");
+            $errors += 1;
+        }
+    }
+
+    record_info('TEST', 'Testing patterns and packages can be installed.');
     for (my $i = 0; $i < (scalar @$tdata); $i++) {
         my $container = "bci-repo-tester$i";
-
         assert_script_run("podman run --name $container -dt $image");
-        # query default setup
-        container_exec($container, 'zypper ref', timeout => 180);
-        container_exec($container, 'zypper lr -d');
-        container_exec($container, 'zypper search -t pattern', timeout => 180);
-        container_exec($container, "zypper -q -s 11 pa --orphaned | tee -a repo.org", timeout => 600);
-        # remove container-suseconnect, in order to install the Cloud patterns
-        container_exec($container, 'zypper -n rm container-suseconnect', timeout => 180);
-        # refresh services in order to remove container-suseconnect-zypp orphaned services
-        container_exec($container, 'zypper refs');
-        # remove SLE_BCI repo pointing to official update servers and add SUT repo
-        container_exec($container, 'zypper rr 1');
-        container_exec($container, "zypper ar http://openqa.suse.de/assets/repo/$bci_repo BCI_TEST");
-        container_exec($container, 'zypper ref', timeout => 180);
-        container_exec($container, 'zypper lr -d');
-        record_info("Patterns", join(', ', @{$tdata->[$i]->{patterns}}));
+        prepare_repo($container);
```

Comment:
I didn't quite get what is the purpose of `prepare_repo()` and why you call it twice. Can you use more specific name or add a comment?

Reply:
It replaces the default repo which points to updates.suse.com to the repo under test in suse.de space. Also, to avoid getting repos from the host, it deletes the container-suseconnect plugin. So, in the end, it "prepares the BCI repo" under test inside the container.